### ISSUE TITLE
Show keypaths!

### ISSUE THREAD

**Issue by lf94** [digitalbitbox/bitbox-wallet-app#1238]
Hey!

So it would be extremely useful if the keypath for an address is shown near it. This makes it less error prone and easier for users to specify in other applications. For example it would make entering the wrong address impossible, and the worst case is the funds are sent to a weird keypath.

**Comment by lf94** [comment]
Nevermind, it appears Trezor doesn't display them either. This is disappointing, but since it's not prominent, I won't request it.

**Comment by x1ddos** [comment]
I think this is actually a useful feature!

**Comment by lf94** [comment]
While it is useful BitBox, would be the only wallet doing it... Would need to maybe converse with the other wallet vendors, mainly Ledger and Trezor. Having a ubiquitous experience I think is important in the cryptocurrency space.

**Comment by benma** [comment]
> and the worst case is the funds are sent to a weird keypath.

This shouldn't happen if you verify the address on the BitBox02, as the BitBox02 performs keypath validation.

Maybe we can add the account-level xpub to the account information page, where the account xpub is shown. Xpubs are already quite technical, I wouldn't mind the keypath to be shown there.

**Comment by wilya7** [comment]
> > and the worst case is the funds are sent to a weird keypath.
> 
> This shouldn't happen if you verify the address on the BitBox02, as the BitBox02 performs keypath validation.
> 
> Maybe we can add the account-level xpub to the account information page, where the account xpub is shown. Xpubs are already quite technical, I wouldn't mind the keypath to be shown there.

Keypath is the derivation path? If yes, I would really like that. Question: is the new feature "multiple accounts" encoded there? In this case it would be very useful to have this information for each account information page), in case a recovery on another wallet becomes necessary.

**Comment by benma** [comment]
@wilya7 agreed, it would be helpful to show this information in the account info :+1:

**Comment by lf94** [comment]
What is BitBox doing to address the educational part of this? It would also be great to see collaboration with Trezor at the very least!

**Comment by benma** [comment]
> What is BitBox doing to address the educational part of this? It would also be great to see collaboration with Trezor at the very least!

What kind of collaboration are you thinking about? I was thinking to simply explain in the in-app guide what it means and link to BIP44.

**Comment by lf94** [comment]
Simply working with them to make keypaths a standard piece of knowledge for regular people to know about.

Linking to any BIP will be simply ignored by the non-tech population. Sure it's good to link it, but do not make it the primary resource for learning what a keypath is.

**Comment by wilya7** [comment]
The x/y/z-pub and the derivation path is the necessary information for a universal wallet recovery.  If the human capital of Shiftcrypto can be spent in addition to developing their products to write a nice guide in the form of a blog post or a help entry, than great. However, if time is a scarce resource and trade-offs have to be made, I would encourage the developers to first make the information available.

**Comment by lf94** [comment]
It's presumed we all know what a keypath is; I'm not sure why it's important to state why someone would need a keypath in this conversation.

The capital Shiftcrypto has to expend on supporting their own product is irrelevant to me. If a feature is implemented but not explained well, it will have little impact and add little value to the overall product.

My suggestions come from a place of trying to maximize the impact and usefulness of the addition. There is benefit to getting other wallet vendors on board with displaying keypaths as a regular piece of information. Keypaths have to be at the forefront of the cryptocurrency vocabulary, not a secondary citizen. Non-technical people barely know what the purpose of a hardware wallet is.

**Comment by wilya7** [comment]
> It's presumed we all know what a keypath is; I'm not sure why it's important to state why someone would need a keypath in this conversation.

We are referring to two different things. I am talking about the fact that when I go to my bitbox, I would like to know the derivation path for each of my accounts, which I just cannot at the moment.

> The capital Shiftcrypto has to expend on supporting their own product is irrelevant to me. If a feature is implemented but not explained well, it will have little impact and add little value to the overall product.

My guess is, it is relevant to them, and since you are asking Shiftcrypto employee to do the work necessary to fulfill your request, their priorities are very relevant in the context of this issue. 

About the feature in question, do you know for sure what is the derivation path of each of your accounts in your bitbox? I don't. I can guess, but I do not know for sure. Without that, I cannot recover my accounts in electrum or samurai wallet or any other wallet. I can think of several scenarios where my funds can be at risk if I cannot do that. So, forgive me if I care more about the information and slightly less about the documentation (which I am all for).

Since I believe I clarified my point and I understood yours, I am not going to argue this issue any further.

**Comment by benma** [comment]
Let's make the info available and add an in-app guide entry with further information (that would be some text and a link). That should give beginners an entry to learn more about it if they desire to do so, and advanced users the information they were looking for.

**Comment by lf94** [comment]
@wilya7 I think you misunderstand almost everything: there is no disagreement here. :slightly_smiling_face: 

Your point is no clearer to me, sorry.

We will see what the result will be from @benma 's work and the impact it has in the following months :slightly_smiling_face:

**Comment by My1** [comment]
> > and the worst case is the funds are sent to a weird keypath.
> This shouldn't happen if you verify the address on the BitBox02, as the BitBox02 performs keypath validation.

well if you have it send to like m/89'/0'/16548856'/0/156453 that is a valid bitcoin path yet it would be a mega pain to find your coins.

**Comment by benma** [comment]
> > > and the worst case is the funds are sent to a weird keypath.
> > > This shouldn't happen if you verify the address on the BitBox02, as the BitBox02 performs keypath validation.
> 
> well if you have it send to like m/89'/0'/16548856'/0/156453 that is a valid bitcoin path yet it would be a mega pain to find your coins.

This keypath wouldn't pass validation in the BitBox02 - we check that it conforms to one of the few standard derivation paths, exactly for the reason you stated: it would be a major pain to find the coins again, and this could be exploited by a bad host to hide your coins.

**Comment by My1** [comment]
> This keypath wouldn't pass validation in the BitBox02 - we check that it conforms to one of the few standard derivation paths

why wouldnt this pass? aside from a high account and index number? I mean unless I messed up the purpose this should be a totally valid BTC path

**Comment by benma** [comment]
> > This keypath wouldn't pass validation in the BitBox02 - we check that it conforms to one of the few standard derivation paths
> 
> why wouldnt this pass? aside from a high account and index number? I mean unless I messed up the purpose this should be a totally valid BTC path

What do you mean by valid BTC path? Any path is technically valid, but we chose to only accept a small subset of keypaths (purpose 89' is not supported, and for the purposes that are supported, only a relatively small number of accounts and addresses are allowed).

**Comment by My1** [comment]
> purpose 89' is not supported

okay then I forgot the actual number of the segwit ones, like 44 doesnt work anyway, I thought 89 was something segwit.



> only a relatively small number of accounts and addresses are allowed

that part of what I guess is important

**Comment by benma** [comment]
Added in https://github.com/digitalbitbox/bitbox-wallet-app/pull/1329

**Comment by thisconnect** [comment]
released in 4.29.0
